donut/modules/calendar/helpers.py

```diff
@@ -0,0 +1,412 @@
+                events = events_result.get('items', [])
+                all_events.extend(events)
+                next_page = 'nextPageToken' in events_result
+        return all_events
+    except:
+        return []
+
+
+def add_event(name,
+              description,
+              calendar_tag,
+              begin_time,
+              end_time,
+              update,
+              location='',
+              event_id=""):
+    """
+    Adds or updates events. If we are updating the event, 
+    we need to have the event_id. 
+    name: name of event
+    description: description of event
+    calendar_tag: a list of calendar tags for an event. Only
+        a single value for updates
+    begin_time: in format YYYY-MM-DDThh:mm:00Z
+    end_time: as above. 
+    update: bool for it we are adding events or updating them
+    location: not required; location for event
+    """
+    try:
+        if update:
+            event = service.events().get(
+                calendarId=cal_id[calendar_tag[0]],
+                eventId=event_id).execute()
+            event['start']['dateTime'] = begin_time
+            event['end']['dateTime'] = end_time
+            event['start']['date'] = None
+            event['end']['date'] = None
+            event['end']['timeZone'] = 'America/Los_Angeles'
+            event['start']['timeZone'] = 'America/Los_Angeles'
+            event['summary'] = name
+            event['description'] = description
+            event['location'] = location
+            if get_permission()[calendar_tag[0]]:
+                new_event = service.events().update(
+                    calendarId=cal_id[calendar_tag[0]],
+                    eventId=event['id'],
+                    body=event).execute()
+                insert_event_to_db(calendar_tag[0], new_event['id'], name,
+                                   description, location, begin_time, end_time,
+                                   flask.session['username'])
+            return ''
+        for cal_tag in calendar_tag:
+            event = {
+                'summary': name,
+                'description': description,
+                'start': {
+                    'dateTime': begin_time,
+                    'timeZone': 'America/Los_Angeles',
+                },
+                'end': {
+                    'dateTime': end_time,
+                    'timeZone': 'America/Los_Angeles',
+                },
+                'location': location,
+                'visibility': 'public',
+            }
+            if get_permission()[cal_tag]:
+                created_event = service.events().insert(
+                    calendarId=cal_id[cal_tag], body=event).execute()
+                insert_event_to_db(cal_tag, created_event['id'], name,
+                                   description, location, begin_time, end_time,
+                                   flask.session['username'])
+        return ''
+    except Exception as e:
+        return e.get('message', e)
```

Comment:
wait why? I wrapped everything in try catch in case of google calendar api failure

Reply:
It's just a little dangerous to hide *all* exceptions, even if you expect some exceptions to be thrown. What if the API becomes permanently unreachable? What if there's a database error? It would be nice for the dev team to get some information about what went wrong